ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,306 @@
+        .format(instance_id, elb_name))
+
+    _add_instance_to_load_balancer_list_in_properties(instance_id)
+
+
+def _add_health_check_to_load_balancer(load_balancer, health_check):
+
+    hc = _create_health_check(health_check)
+
+    try:
+        load_balancer.configure_health_check(hc)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Health check not added to Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Health check added to Load Balancer {0}.'
+        .format(load_balancer.name))
+
+
+@operation
+def use_external_elb(**_):
+
+    utils.use_external_resource(ctx.node.properties)
+    # utils.get_external_resource_id_or_raise(
+    #     'elb_name', ctx.target.instance)
+    #
+    # _get_existing_load_balancer()
+
+
+@operation
+def delete_load_balancer(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        utils.unassign_runtime_property_from_resource(
+            constants.EXTERNAL_RESOURCE_ID, ctx.instance)
+        return False
+
+    ctx.logger.debug('Attempting to delete Load Balancer.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    try:
+        lb = _get_existing_load_balancer(elb_name)
+        lb.delete()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer {0} not deleted. '
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return True
+
+
+def _create_health_check(properties):
+
+    health_check = {
+        'interval': constants.HEALTH_CHECK_INTERVAL,
+        'healthy_threshold': constants.HEALTH_CHECK_HEALTHY_THRESHOLD,
+        'timeout': constants.HEALTH_CHECK_TIMEOUT,
+        'unhealthy_threshold': constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+    }
+
+    for key, value in properties.iteritems():
+        health_check[key] = value
+
+    health_check = HealthCheck(**health_check)
+    return health_check
+
+
+def _create_load_balancer_params():
+    params_dict = {}
+    params_dict['listeners'] = ctx.node.properties['listeners']
+    params_dict['name'] = ctx.node.properties['elb_name']
+    params_dict['zones'] = ctx.node.properties['zones']
+    if 'security_groups' in ctx.node.properties.keys():
+        params_dict['security_groups'] = ctx.node.properties['security_groups']
+    return params_dict
+
+
+def _create_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to Create Load Balancer.')
+
+    params_dict = _create_load_balancer_params()
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
```

Comment:
Why not? it is access to an external resource (boto configuration) and therefore can fail even if the code is fine.